optimum/intel/openvino/quantization.py

```diff
@@ -252,45 +247,39 @@ def quantize(
         >>> from transformers import AutoModelForCausalLM
         >>> model = AutoModelForCausalLM.from_pretrained("databricks/dolly-v2-3b")
         >>> quantizer = OVQuantizer.from_pretrained(model, task="text-generation")
-        >>> quantizer.quantize(save_directory="./quantized_model", weights_only=True)
+        >>> ov_config = OVConfig(quantization_config=OVWeightQuantizationConfig(bits=8, sym=True))
+        >>> quantizer.quantize(ov_config=ov_config, save_directory="./quantized_model")
         >>> optimized_model = OVModelForCausalLM.from_pretrained("./quantized_model")
         ```
         """
+        if "calibration_dataset" in kwargs:
+            raise ValueError(
+                "`calibration_dataset` argument is deprecated. Please provide calibration dataset "
+                "with `ov_config.quantization_config.dataset`."
+            )
+        if "weights_only" in kwargs:
+            raise ValueError(
+                "`weights_only` argument is deprecated. Please provide `ov_config.quantization_config` "
```

Comment:
might make sense to leave at least one or two minor release to warn people about this before deprecating it, what do you think ?

Reply:
Replaced raising `ValueError` by giving warning. This should be enough since if `ov_config.quantization_config` is omitted, weight compression is applied.